packages/graphql-transformer-core/src/util/sanity-check.ts

```diff
@@ -75,7 +75,9 @@ export const sanityCheckDiffs = (
  */
 export const getCantEditKeySchemaRule = (iterativeUpdatesEnabled = false) => {
   const cantEditKeySchemaRule = (diff: Diff): void => {
-    if (diff.kind === 'E' && diff.path.length === 8 && diff.path[5] === 'KeySchema') {
+    const sortKeyAdded = diff.kind === 'A' && diff.path.length === 6 && diff.path[5] === 'KeySchema' && diff.index === 1;
+    const keySchemaModified = diff.kind === 'E' && diff.path.length === 8 && diff.path[5] === 'KeySchema';
```

Comment:
Does this cover removing the existing sort key?

Reply:
it does, I adjusted the variable naming appropriately